src/party.cpp

```diff
@@ -11,15 +11,15 @@
 
 extern Game g_game;
 
-Party::Party(Player* leader) : leader(leader) { leader->setParty(this); }
+Party::Party(std::shared_ptr<Player> leader) : leader(leader) { leader->setParty(this); }
 
 void Party::disband()
 {
 	if (!tfs::events::party::onDisband(this)) {
 		return;
 	}
 
-	Player* currentLeader = leader;
+	std::shared_ptr<Player> currentLeader = leader;
```

Comment:
Could this be `const auto`
```suggestion
	const auto currentLeader = leader;
```